library/src/androidTest/java/org/xmtp/android/library/ClientTest.kt

```diff
@@ -967,4 +976,81 @@ class ClientTest {
         val logFiles2 = Client.getXMTPLogFilePaths(context)
         assertEquals(logFiles2.size, 0)
     }
+
+    @Test
+    fun testNetworkDebugInformation() = runBlocking {
+        val key = SecureRandom().generateSeed(32)
+        val context = InstrumentationRegistry.getInstrumentation().targetContext
+        val alixWallet = PrivateKeyBuilder()
+        val alix = Client.create(
+            account = alixWallet,
+            options = ClientOptions(
+                ClientOptions.Api(XMTPEnvironment.LOCAL, false),
+                appContext = context,
+                dbEncryptionKey = key
+            )
+        )
+        delay(2000)
+        val aggregateStats1 = alix.XMTPDebugInformation.aggregateStatistics
+        println("Aggregate Stats Create:\n$aggregateStats1")
+
+        val apiStats1 = alix.XMTPDebugInformation.apiStatistics
+        assertEquals(1, apiStats1.uploadKeyPackage)
+        assertEquals(0, apiStats1.fetchKeyPackage)
+        assertEquals(2, apiStats1.sendGroupMessages)
+        assertEquals(0, apiStats1.sendWelcomeMessages)
+        assertEquals(4, apiStats1.queryGroupMessages)
+        assertEquals(0, apiStats1.queryWelcomeMessages)
+        assertEquals(0, apiStats1.subscribeMessages)
+        assertEquals(0, apiStats1.subscribeWelcomes)
+
+        val identityStats1 = alix.XMTPDebugInformation.identityStatistics
+        assertEquals(1, identityStats1.publishIdentityUpdate)
+        assertEquals(3, identityStats1.getIdentityUpdatesV2)
+        assertEquals(2, identityStats1.getInboxIds)
+        assertEquals(0, identityStats1.verifySmartContractWalletSignature)
+
+        val job = CoroutineScope(Dispatchers.IO).launch {
+            alix.conversations.streamAllMessages().collect { }
+        }
+        val group = alix.conversations.newGroup(emptyList())
+        group.send("hi")
+
+        val aggregateStats2 = alix.XMTPDebugInformation.aggregateStatistics
+        println("Aggregate Stats Create:\n$aggregateStats2")
+
+        val apiStats2 = alix.XMTPDebugInformation.apiStatistics
+        assertEquals(1, apiStats2.uploadKeyPackage)
+        assertEquals(0, apiStats2.fetchKeyPackage)
+        assertEquals(6, apiStats2.sendGroupMessages)
+        assertEquals(0, apiStats2.sendWelcomeMessages)
+        assertEquals(11, apiStats2.queryGroupMessages)
+        assertEquals(1, apiStats2.queryWelcomeMessages)
+        assertEquals(1, apiStats2.subscribeMessages)
```

Comment:
I seem to fail on the `subscribeMessages` line. added to post release issues to look at

Reply:
O I think I might know why. I think clearing the stats before running this test may help here. TBD.